### DISCUSSION TITLE
arch linux automated package and build

### DISCUSSION THREAD

**Discussion by Bewn** [oobabooga/text-generation-webui#1932] [Show and tell]
I’ve written a pacman style PKGBUILD for this project and it’s dependencies which you can find [here](https://github.com/Bewn/textgen-ben) in the bendir/build directory. 
I’m using micromamba to manage python and only requiring git and wget as hard dependencies, it’s a simple `makepkg -si` install through pacman. Can also be converted to a bash script for use with any distro, which I’ll do in a bit, or even run in a container. The point is it will be a one-click way to get to a running gradio client of oobabooga’s main code. I’ll be building an interface on top for my own interest but will maintain compat with main branch in case of useful merge potential!

edit:
also have added a generic install for any distro with a unix shell
at [this repo](https://github.com/Bewn/textgen-one-click)

**Comment by Bewn** [comment]
note: right now it’s just for testing, it will build a portable folder in your home directory which will contain the server runner. right now its just installing all python dependencies

**Reply by luthis1124** [reply]
Freakin excellent, I am having silly issues with the installer on the main page. Any assistance I can provide?

**Reply by Bewn** [reply]
feel free to clone either of the repos, one has the arch build and one is a one click generic shell script which should work on any distro, note that it will change a lot in the next couple days so keep up to date or wait a little bit till i put up a stable release!

**Reply by Bewn** [reply]
as far as assistance I’d say once I have it set up just test running it on your machine, distros other than arch for the shell script will need testing, I’ll set up some containers when I get time

**Comment by eucdee** [comment]
does it support AMD ROCm?
i gave my Arch Distro another chance today but only StableDiffusion figuered it out..
maybe tomorrow ill give it another shot...

**Reply by Bewn** [reply]
I’m trying to make it as generic as possible so I think it should work with rocm as an install but i’m pretty sure upstream depends on some cuda code so it might need a bit of modification to actually run the llms and stuff, I’ll look into it but my old amd card broke and i’m running a 3070Ti

**Reply by Bewn** [reply]
as of latest commit I added a build-time check for rocm or cuda to install the right pytorch, so that should be usable!
